### PR TITLE
Update clang-format to 17.0.3 and port clang-format actions from obs-studio

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

This PR does a few things:
1. Port the clang-format Github Actions and Workflow from the obs-studio repo
2. Port the .clang-format file/settings from the obs-studio repo
3. Applies clang-format changes from clang-format 17 and the new .clang-format settings

Some of these commits can and probably should be squashed so that there is not a commit that would fail CI. Alternatively, we rearrange the commits so that the old workflow is removed first, and then the files are formatted, and then new workflow is added to validate.

### Motivation and Context

Have been wanting to do this for a while for consistency and finally get obs-browser off of clang-format 13.0.1.

### How Has This Been Tested?

Ran clang-format locally. Will see if CI passes.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by PatTheMav — APPROVED** [review]

**Review by WizardCM — APPROVED** [review]
The output of this PR matches what I expected when I originally performed formatting with the wider column layout.

**Comment by WizardCM** [comment]
I've rebased this PR to ensure it still passes with the other changes to master. Will leave up to @RytoEX to squash as necessary & merge.

After this PR is merged, it'd also be worth committing a `.git-blame-ignore-revs` with commits from this PR and my original one.

**Pull request merged**
